skopt/plots.py

```diff
@@ -88,3 +90,157 @@ def plot_convergence(*args, **kwargs):
         ax.legend(loc="best")
 
     return ax
+
+
+def _format_scatter_plot_axes(ax, space):
+    # Deal with formatting of the axes
+    for i in range(space.n_dims):
+        for j in range(space.n_dims):
+            ax_ = ax[i, j]
+
+            if j > i:
+                ax_.axis("off")
+
+            # adjust bounds for every off-diagonal axis
+            if i != j:
+                ax[i, j].set_ylim(*space.dimensions[i].bounds)
+                ax[i, j].set_xlim(*space.dimensions[j].bounds)
+
+            ax_.xaxis.set_major_locator(MaxNLocator(6, prune='both'))
+            ax_.yaxis.set_major_locator(MaxNLocator(6, prune='both'))
+
+            if i < space.n_dims - 1:
+                ax_.set_xticklabels([])
+            # bottom row
+            else:
+                [l.set_rotation(45) for l in ax_.get_xticklabels()]
+
+            if j > 0:
+                ax_.set_yticklabels([])
+
+    return ax
+
+
+def plot_objective_function(result, levels=10):
+    """Pairwise scatter plot of objective function
+
+    Pairwise scatter plots are shown on the off-diagonal for each
+    dimension of the search space. A red point indicates the minimum.
+
+    Note: the objective function contours are obtained by interpolating
+    between samples. The surrogate model is not used.
+
+    Note: search spaces that contain `Categorical` dimensions are
+    currently not supported by this function.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The result for which to create the scatter plot matrix.
+
+    * `levels` [int, default=10]
+        Number of levels to draw on the contour plot, passed directly
+        to `plt.contour()`
+
+    Returns
+    -------
+    * `ax`: [`Axes`]:
+        The matplotlib axes.
+    """
+    space = result.space
+    samples = np.asarray(result.x_iters)
+    order = range(samples.shape[0])
+    rvs = space.rvs(n_samples=10)
```

Comment:
Right now I tuned the numbers so that they aren't super painful when you want to plot often. I had started with `100` points but this was super slow. +1 for 100 as default and we should definitely make it a parameter that can be adjusted by the user.